core/modules/node/node.pages.inc

```diff
@@ -374,26 +385,8 @@ function node_form($form, &$form_state, Node $node) {
 }
 
 /**
- * Form submission handler for the 'Delete' button for node_form().
```

Comment:
Deleting this seems like the right thing to do.  I need to triple check that this is not called anywhere else and check the validation function logic to see if there is any logic in there that is desirable on delete op.

PR is looking really good!